### DISCUSSION TITLE
Is --resync is really required with non-sync options?

### DISCUSSION THREAD

**Discussion by bpozdena** [abraunegg/onedrive#2101] [General]
I was just curious whether there is any technical reason behind enforcing `--resync` when I try to use the OneDrive client to just obtain `--list-shared-folders`  or perform other non-sync options. 

I came across this when I was trying to get updated list of shared folders shortly after I made change to `business_shared_folders` file. I then had to wait an hour for the resync to complete in order to run the `--list-shared-folders` command again. 

```
onedrive --confdir='/home/bob/.config/onedrive/accounts/bpozdena@company.com' -v --list-shared-folders  
Using 'user' Config Dir: /home/bob/.config/onedrive/accounts/bpozdena@company.com
Using config option for Global Azure AD Endpoints
Configuration file successfully loaded
An application configuration change has been detected where a --resync is required
```

```
onedrive --confdir='/home/bob/.config/onedrive/accounts/bpozdena@company.com' -v --get-O365-drive-id '*' 
Using 'user' Config Dir: /home/bob/.config/onedrive/accounts/bpozdena@company.com
Using config option for Global Azure AD Endpoints
Configuration file successfully loaded
An application configuration change has been detected where a --resync is required
```

```
onedrive --confdir='/home/bob/.config/onedrive/accounts/bpozdena@company.com' -v --get-file-link test
Using 'user' Config Dir: /home/bob/.config/onedrive/accounts/bpozdena@company.com
Using config option for Global Azure AD Endpoints
Configuration file successfully loaded
An application configuration change has been detected where a --resync is required
```

**Comment by abraunegg** [comment]
@bpozdena 
> Is --resync is really required with non-sync options?

The trigger for this was the updated `business_shared_folders` file and the checking of 'has something changed' before continuing is taking priority here.

This being said - all those options should not be impacted by any config change - as, they do not do any sync.

Please open a bug report.

**Reply by bpozdena** [reply]
Thanks for the clarification. Reported in #2102 .
